crates/oxc_semantic/src/builder.rs

```diff
@@ -106,9 +106,10 @@ impl<'a> SemanticBuilder<'a> {
     }
 
     #[must_use]
-    pub fn with_trivias(mut self, trivias: &Rc<Trivias>) -> Self {
-        self.trivias = Rc::clone(trivias);
-        self.jsdoc = JSDocBuilder::new(self.source_text, trivias);
+    pub fn with_trivias(mut self, trivias: Trivias) -> Self {
+        let trivias = Rc::new(trivias);
+        self.trivias = Rc::clone(&trivias);
+        self.jsdoc = JSDocBuilder::new(self.source_text, &trivias);
```

Comment:
I'm not sure this PR fixes #708 eventually. The `Rc` still lives on `Semantic`, which makes `Semantic` not `Send`.

If you have a bunch of `Semantic`s and want to apply `rayon` on them, the rustc would still complain that `Semantic` is not `Send` or `Sync`.